crypto/dh/dh_gen.c

```diff
@@ -126,5 +126,7 @@ static int dh_builtin_genparams(DH *ret, int prime_len, int generator,
         BN_CTX_end(ctx);
         BN_CTX_free(ctx);
     }
+    if(ret->p!=NULL)BN_free(ret->p);
+    if(ret->g!=NULL)BN_free(ret->g);
```

Comment:
These changes are wrong and will result in a double free because `ret->p` will continue to point to the allocated memory. When the DH object gets freed this will fail.